influxql/select.go

```diff
@@ -822,18 +822,40 @@ func buildTransformIterator(lhs Iterator, rhs Iterator, op Token, ic IteratorCre
 			left:  newBufIntegerIterator(left),
 			right: newBufIntegerIterator(right),
 			fn: func(a *IntegerPoint, b *IntegerPoint) *FloatPoint {
-				p := &FloatPoint{
-					Name: a.Name,
-					Tags: a.Tags,
-					Time: a.Time,
-					Aux:  a.Aux,
-				}
-				if (a != nil && b != nil) && (!a.Nil && !b.Nil) {
-					p.Value = fn(a.Value, b.Value)
+				if a != nil && b != nil {
```

Comment:
This seems a bit clunky. Possibly it would be clearer if you did:

```go
if a == nil {
	return &FloatPoint{.....}
} else if b == nil {
	return &FloatPoint{.....}
}
p := &FloatPoint{
	Name: a.Name,
	Tags: a.Tags,
	Time: a.Time,
	Aux:  a.Aux,
}
if !a.Nil && !b.Nil {
	p.Value = fn(a.Value, b.Value)
	return p
} else {
	p.Value = float64(0)
	p.Nil = true
	return p
}
```

that way less code is indented.